asmcomp/cmm.mli

```diff
@@ -134,10 +134,10 @@ and expression =
   | Clet of Ident.t * expression * expression
   | Cassign of Ident.t * expression
   | Ctuple of expression list
-  | Cop of operation * expression list
+  | Cop of operation * expression list * Debuginfo.t
   | Csequence of expression * expression
   | Cifthenelse of expression * expression * expression
-  | Cswitch of expression * int array * expression array
+  | Cswitch of expression * int array * expression array * Debuginfo.t
```

Comment:
I think that stepping through the mutation may not be possible. This is due to missing features of DWARF -- although there are proposals to improve things in DWARF 6.

Basically, when you optimise something out of a program, DWARF should provide you with a way to artificially put it back in again. For example, it lets you describe values which have been optimised away. However, the stepping behaviour is not directly specifiable in DWARF yet, instead it just runs the program until the line number associated with the current pc changes, so there is no way to tell it about an operation (e.g. the mutation of an optimised local variable) which has been removed. (This is also what prevents expression level stepping in gdb and lldb).